generator/src/main/java/org/apache/kafka/message/JsonConverterGenerator.java

```diff
@@ -53,17 +53,17 @@ public void generateAndWrite(MessageSpec message, BufferedWriter writer)
         buffer.incrementIndent();
         generateConverters(message.dataClassName(), message.struct(),
             message.validVersions());
+
+        for (Iterator<StructSpec> iter = structRegistry.commonStructs(); iter.hasNext(); ) {
+            generateStructClass(iter.next(), message.validVersions());
+        }
+
         for (Iterator<StructRegistry.StructInfo> iter = structRegistry.structs();
```

Comment:
It's not clear whether we need to generate the common struct JSON converters first.  Can you try without (or explain why it needs to be with?)  Or is the iterator returned from `structs` not returning the correct data for common structs?